proxmoxtf/resource_virtual_environment_vm.go

```diff
@@ -1163,10 +1174,52 @@ func resourceVirtualEnvironmentVMCreateClone(ctx context.Context, d *schema.Reso
 
 	cloneTimeout := d.Get(mkResourceVirtualEnvironmentVMTimeoutClone).(int)
 
-	if cloneNodeName != "" && cloneNodeName != nodeName {
+	datastoreStatus, err := veClient.GetDatastoreStatus(ctx, cloneNodeName, cloneDatastoreID)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Whether
+	cloneDatastoreShared := proxmox.CustomBool(false)
+	if datastoreStatus.Shared != nil {
+		cloneDatastoreShared = *datastoreStatus.Shared
+	}
+
+	if cloneNodeName != "" && cloneNodeName != nodeName && cloneDatastoreShared {
+		// If the source and the target node are not the same, only clone directly to the target node if
+		//  the target datastore is shared. Directly cloning to non-shared storage on a different node is
+		//  currently not supported by proxmox.
 		cloneBody.TargetNodeName = &nodeName
 
 		err = veClient.CloneVM(ctx, cloneNodeName, cloneVMID, cloneRetries, cloneBody, cloneTimeout)
+	} else if cloneNodeName != "" && cloneNodeName != nodeName && !cloneDatastoreShared {
+		// If the source and the target node are not the same and the target datastore is not shared, clone
+		//  to the source node and then migrate to the target node. This is a workaround for missing functionality
+		//  in the proxmox api as recommended per https://forum.proxmox.com/threads/500-cant-clone-to-non-shared-storage-local.49078/#post-229727
+		cloneBody.TargetStorage = &cloneTempDatastoreID
+
+		// Temporarily clone to local node
+		err = veClient.CloneVM(ctx, cloneNodeName, cloneVMID, cloneRetries, cloneBody, cloneTimeout)
+
+		if err != nil {
```

Comment:
Just a small style thing, I would keep error check right on the next line after the call that produced the error, here and line 1178. 
I think there is already a linting rule for that, but I have most of them disabled at the moment. You know, the original code is not super great.